index.js

```diff
@@ -1,26 +1,30 @@
 const core = require('@actions/core');
-
-const { execSync } = require('child_process');
+const io = require('@actions/io');
+const exec = require('@actions/exec');
 
 const host = core.getInput('host');
 const port = core.getInput('port');
 const socketPath = core.getInput('socket-path');
 const key = core.getInput('key');
+const lifetimeInSeconds = core.getInput('lifetime_in_seconds');
 
-console.log(`Attempting to create ${socketPath}...`);
-try {
-    execSync(
-        'mkdir -p ~/.ssh && ' +
-        `ssh-keyscan${port ? ` -p ${port}` : ''} "${host}" >> ~/.ssh/known_hosts && ` +
-        `eval $(ssh-agent -a "${socketPath}") && ` +
-        `echo "${key}" | base64 -d | ssh-add -`
-    );
-} catch (e) {
-    console.error(e.message);
-    process.exit(1);
-}
+async function run() {
+    console.log('checking if ~/.ssh is created...');
+    await io.mkdirP('~/.ssh');
+
+    console.log('ssh-keyscan the provided domain...');
+    let portCommand = port ? `-p ${port}` : '';
+    await exec.exec('ssh-key', [portCommand, host, '>> ~/.ssh/known_hosts']);
 
-console.log(`Created ${socketPath}`);
-core.setOutput('socket-path', socketPath);
+    console.log(`Attempting to create if not found. ${socketPath}...`);
+    await exec.exec(`eval $(ssh-agent -a "${socketPath}")`);
+
+    console.log('Attempting to add key to agent...');
+    await exec.exec(`echo "${key}" | base64 -d | ssh-add -t ${lifetimeInSeconds} -`);
+
+    console.log(`Created ${socketPath}`);
+    core.setOutput('socket-path', socketPath);
+    console.log('Done; exiting.');
+}
 
-console.log('Done; exiting.');
+run();
```

Comment:
Does github actions know to wait for async things? I'm assuming you wrapped this in a function so you could use `await`? I think we might to handle errors and do `core.setFailed`. Also probably use the core output fns instead of `console.log`.

Reply:
I think they do. I pulled it from the example - https://github.com/actions/javascript-action/blob/main/index.js